component/prometheus/interceptor.go

```diff
@@ -102,13 +103,11 @@ func (a *interceptappender) Append(ref storage.SeriesRef, l labels.Labels, t int
 	if ref == 0 {
 		ref = storage.SeriesRef(a.ls.GetOrAddGlobalRefID(l))
 	}
-
-	if value.IsStaleNaN(v) {
-		a.ls.AddStaleMarker(uint64(ref), l)
-	} else {
-		// Tested this to ensure it had no cpu impact, since it is called so often.
-		a.ls.RemoveStaleMarker(uint64(ref))
-	}
+	a.stalenessTrackers = append(a.stalenessTrackers, labelstore.StalenessTracker{
```

Comment:
We create a `labelstore.StalenessTracker` here, but later in `func (s *service) TrackStaleness(ids []StalenessTracker)` we convert these to `&staleMarker{}` and calculate the labels hash. Could we instead create the `&staleMarker{}` type right away here and calculate the hash here? 

That way there will be less work and structs to create, the code will get simpler too I think.

Reply:
I feel those are owned by two different things. Mainly the last marked state should really only be set by the labelstore itself and exposing that field feels off. This gets cleaned up slightly in the next PR.